modules/internal/ChapelRange.chpl

```diff
@@ -667,13 +669,61 @@ module ChapelRange {
   }
   
   
-  inline proc chpl_by_help(r: range(?i,?b,?s), step) {
+  proc chpl_check_step_integral(step) {
+    if !isIntegral(step.type) then
+      compilerError("can't apply 'by' using step of a non-integral type ",
+                    typeToString(step.type));
+  }
+  
+  proc chpl_need_to_check_step(step, type strType) param {
```

Comment:
Might be a personal preference, but this doesn't really seem worthwhile to me, especially since check will be off with --fast anyways.